core/server/worker/src/test/java/alluxio/worker/block/BlockMasterClientTest.java

```diff
@@ -77,4 +121,314 @@ public void convertBlockListMapToProtoMergeDirsInSameTier() {
+            .withMaxDuration(Duration.of(500, ChronoUnit.MILLIS))
+            .withInitialSleep(Duration.of(0, ChronoUnit.MILLIS))
+            .withMaxSleep(Duration.of(100, ChronoUnit.MILLIS))
+            .withSkipInitialSleep()
+            .build());
+  }
+
+  public void register(boolean stream) throws Exception {
+    final long workerId = 0L;
+    final List<String> storageTierAliases = ImmutableList.of("MEM");
+    final Map<String, Long> totalBytesOnTiers = ImmutableMap.of("MEM", 1024 * 1024L);
+    final Map<String, Long> usedBytesOnTiers = ImmutableMap.of("MEM", 1024L);
+    final Map<BlockStoreLocation, List<Long>> currentBlocksOnLocation = ImmutableMap.of();
+    final Map<String, List<String>> lostStorage = ImmutableMap.of("MEM", ImmutableList.of());
+    final List<ConfigProperty> configList = ImmutableList.of();
+
+    List<Long> registeredWorkerIds = new ArrayList<>();
+
+    Channel mockChannel = createMockService(
+        new BlockMasterWorkerServiceGrpc.BlockMasterWorkerServiceImplBase() {
+          @Override
+          public void registerWorker(RegisterWorkerPRequest request,
+                                     StreamObserver<RegisterWorkerPResponse> responseObserver) {
+            registeredWorkerIds.add(request.getWorkerId());
+            responseObserver.onNext(RegisterWorkerPResponse.newBuilder().build());
+            responseObserver.onCompleted();
+          }
+
+          @Override
+          public StreamObserver<RegisterWorkerPRequest> registerWorkerStream(
+              StreamObserver<RegisterWorkerPResponse> responseObserver) {
+            return new StreamObserver<RegisterWorkerPRequest>() {
+              @Override
+              public void onNext(RegisterWorkerPRequest value) {
+                registeredWorkerIds.add(value.getWorkerId());
+                responseObserver.onNext(RegisterWorkerPResponse.newBuilder().build());
+              }
+
+              @Override
+              public void onError(Throwable t) {
+              }
+
+              @Override
+              public void onCompleted() {
+                responseObserver.onCompleted();
+              }
+            };
+          }
+        });
+
+    BlockMasterClient client = new MockStubBlockMasterClient(
+        MasterClientContext.newBuilder(ClientContext.create(mConf)).build(),
+        mockChannel
+    );
+
+    if (stream) {
+      client.registerWithStream(
+          workerId,
+          storageTierAliases,
+          totalBytesOnTiers,
+          usedBytesOnTiers,
+          currentBlocksOnLocation,
+          lostStorage,
+          configList
+      );
+    } else {
+      client.register(
+          workerId,
+          storageTierAliases,
+          totalBytesOnTiers,
+          usedBytesOnTiers,
+          currentBlocksOnLocation,
+          lostStorage,
+          configList);
+    }
+
+    assertEquals(1, registeredWorkerIds.size());
+    assertEquals(workerId, (long) registeredWorkerIds.get(0));
```

Comment:
Verify other arguments passed to this method, e.g. `currentBlocksOnLocation` is the same.

Reply:
Fixed in new commit